java/vector/src/main/java/org/apache/arrow/vector/stream/MessageSerializer.java

```diff
@@ -290,9 +307,10 @@ public static ArrowBlock serialize(WriteChannel out, ArrowDictionaryBatch batch)
 
     // write the embedded record batch
     long bufferLength = writeBatchBuffers(out, batch.getDictionary());
+    assert bufferLength % 8 == 0;
 
     // Metadata size in the Block account for the size prefix
-    return new ArrowBlock(start, metadataLength + 4, bufferLength + 8);
```

Comment:
I'm not sure - there doesn't seem to be any reason to add 8 to the length that I could tell (although it would just result in some extra padding I think)